src/D2L.CodeStyle.Analyzers/Async/Generator/AsyncToSyncMethodTransformer.cs

```diff
@@ -369,9 +371,13 @@ private VariableDesignationSyntax Transform( VariableDesignationSyntax des )
 			return SyntaxFactory.ParseExpression( $"Task.Run(() => {invocationExpr}).Result" );
 		}
 
-		return invocationExpr
+		invocationExpr = invocationExpr
 			.WithExpression( Transform( invocationExpr.Expression ) )
 			.WithArgumentList( TransformAll( invocationExpr.ArgumentList, Transform ) );
+
+		m_removeAsyncAnywhere = prevRemoveAsyncAnywhereState;

```

Comment:
There are a few exit points in this function where it doesn't get restored.
One way to deal with that is `try`+`finally`:

```
bool prevRemoveAsyncAnywhereState = m_removeAsyncAnywhere;

try {
  // ...
finally {
  m_removeAsyncAnywhere = prevRemoveAsyncAnywhereState;
}
```

no matter how you get out of the `try` block (even exceptions, which is maybe the primary use case for `finally` admittedly (but we don't care)) the `finally` block will execute.